src/drivers/fluid_coreaudio.c

```diff
@@ -114,6 +115,75 @@ get_num_outputs(AudioDeviceID deviceID)
     return total;
 }
 
+void
+set_channel_map(AudioUnit outputUnit, int audio_channels, const char *map_string)
+{
+    OSStatus status;
+    long int number_of_channels;
+    int i, *channel_map;
+    UInt32 property_size;
+    Boolean writable = false;
+
+    status = AudioUnitGetPropertyInfo(outputUnit,
+                                      kAudioOutputUnitProperty_ChannelMap,
+                                      kAudioUnitScope_Output,
+                                      0,
+                                      &property_size, &writable);
+    if(status != noErr)
+    {
+        FLUID_LOG(FLUID_ERR, "Failed to get the channel map size. Status=%ld\n", (long int) status);
+        return;
+    }
+
+    number_of_channels = property_size / sizeof(int);
+    if(!number_of_channels)
+    {
+        return;
+    }
+
+    channel_map = FLUID_ARRAY(int, number_of_channels);
+    if(channel_map == NULL)
+    {
+        FLUID_LOG(FLUID_ERR, "Out of memory.\n");
+        return;
+    }
+
+    FLUID_MEMSET(channel_map, 0, property_size);
```

Comment:
You are zero-initing the map here. If the CoreAudio device provides 6 output channels, but the user has only supplied a list with two entries, outputs 2,3,4,5 would all be mapped to the first left mono channel. Is this intended? Shouldn't you better init with `0xFF`?

Reply:
Good call.

Memsetting this array is primarily being defensive, since the following call to _AudioUnitGetProperty_ fills it out with the default mapping from the OS. I've only ever seen that be 0,1,2,3...n-1 prior to being set, but I don't know if that is guaranteed.

Silence is definitely the better option in the remote chance CoreAudio fails to fill the buffer without returning an error code though. In testing this out, I learned the hard way that it will happily allow you to set an invalid channel number, and the resulting sound is... unpleasant.